flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/AbstractFileStoreWrite.java

```diff
@@ -220,7 +227,9 @@ public void close() throws Exception {
             }
         }
         writers.clear();
-        compactExecutor.shutdownNow();
+        if (lazyCompactExecutor != null) {
+            lazyCompactExecutor.shutdownNow();
```

Comment:
`lazyCompactExecutor` sets to null?

Reply:
A exception should be threw if the write is used again.